pkg/schemamutation/walker.go

```diff
@@ -48,14 +64,18 @@ func (w *mutatingReferenceWalker) walkSchema(schema *spec.Schema) *spec.Schema {
 		}
 	}
 
-	if r := w.walkRefCallback(&schema.Ref); r != &schema.Ref {
+	// Always callback on the whole schema first so that other callbacks
```

Comment:
The callbacks are pre-order DFS? That should probably be in the godoc of the walker of interface (and the call order is needs to be part of the interface contract).

Reply:
This is a DFS but callbacks are not recursive. The recursive part is with `WalkSchema` down below.

Added comments about the call ordering in the type and impl.